include/dxc/dxcapi.internal.h

```diff
@@ -44,10 +44,13 @@ enum LEGAL_INTRINSIC_TEMPLATES {
   LITEMPLATE_MATRIX = 3, // Matrix types (eg. float3x3).
   LITEMPLATE_ANY =
       4, // Any one of scalar, vector or matrix types (but not object).
-  LITEMPLATE_OBJECT = 5, // Object types.
-  LITEMPLATE_ARRAY = 6,  // Scalar array.
+  LITEMPLATE_OBJECT = 5,      // Object types.
+  LITEMPLATE_ARRAY = 6,       // Scalar array.
+  LITEMPLATE_SCALAR_ONLY = 7, // Uncastable scalar types.
+  LITEMPLATE_VECTOR_ONLY = 8, // Uncastable vector types (eg. float3).
+  LITEMPLATE_MATRIX_ONLY = 9, // Uncastable matrix types (eg. float3x3).
```

Comment:
Castability here refers to the ability to truncate or splat something to change it's "shape" (or template or layout, we use a few terms) and not the castability of the contents.